type implementing interface check fails

The following program `sample.go` triggers an unexpected error:

```go
package main

import (
    "fmt"
    "io"
)   
    
func main() {
    x := io.LimitedReader{}
    y := io.Reader(&x)
    fmt.Println(y)
}
```

Expected result:
```console
$ go run ./sample.go
&{<nil> 0}
```

Got:
```console
$ yaegi ./sample.go
run: ./sample.go:10:7: type *io.LimitedReader does not implement interface io.Reader
```

--- a/_test/issue-1136.go
+++ b/_test/issue-1136.go
@@ -1,0 +1,12 @@
+package main
+
+import (
+	"fmt"
+	"io"
+)
+
+func main() {
+	x := io.LimitedReader{}
+	y := io.Reader(&x)
+	fmt.Println(y)
+}

--- a/interp/type.go
+++ b/interp/type.go
@@ -1558,7 +1558,7 @@ func (t *itype) frameType() (r reflect.Type) {
 }
 
 func (t *itype) implements(it *itype) bool {
-	if t.cat == valueT {
+	if isBin(t) {
 		return t.TypeOf().Implements(it.TypeOf())
 	}
 	return t.methods().contains(it.methods())
@@ -1728,6 +1728,17 @@ func isInterfaceBin(t *itype) bool {
 
 func isInterface(t *itype) bool {
 	return isInterfaceSrc(t) || t.TypeOf() != nil && t.TypeOf().Kind() == reflect.Interface
+}
+
+func isBin(t *itype) bool {
+	switch t.cat {
+	case valueT:
+		return true
+	case aliasT, ptrT:
+		return isBin(t.val)
+	default:
+		return false
+	}
 }
 
 func isStruct(t *itype) bool {

